src/plugins/vis_type_vega/public/data_model/types.ts

```diff
@@ -0,0 +1,202 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SearchResponse } from 'elasticsearch';
+import { AggParamsMapping, Filter } from 'src/plugins/data/public';
+import { DslQuery } from 'src/plugins/data/common';
+import { EsQueryParser } from './es_query_parser';
+import { EmsFileParser } from './ems_file_parser';
+import { UrlParser } from './url_parser';
+
+interface Body {
+  aggs?: AggParamsMapping;
```

Comment:
I see you exposed `AggParamsMapping` from the `data` plugin to use here, but I don't think this is the type you want. I believe the vega spec will accept any valid ES query, and that's definitely not what `AggParamsMapping` represents -- it simply represents the params available from the agg configs service, which vega doesn't use.

Perhaps `SearchParams['body']['aggs']` from the `elasticsearch` module is what you'd want? Currently `SearchParams['body']` is `any` so you aren't really getting type safety, but I think that's more likely to be what you are looking for.

Either way we should revert making `AggParamsMapping` public unless it is absolutely necessary.

Reply:
Thanks for the review. I will change it. From the output, It was similar to `AggParamsMapping` so that I added it there. 😅

Reply:
Thanks! And no worries, they are quite similar so I completely understand why you used it here 🙂